src/components/EditCoursePage/index.jsx

```diff
@@ -298,32 +313,18 @@ class EditCoursePage extends React.Component {
   }
 
   formatLocationRestrictionFields(courseData) {
-    if (courseData?.location_restriction) {
-      return {
-        restriction_type: courseData.location_restriction.restriction_type,
-        countries: courseData.location_restriction.countries,
-        states: courseData.location_restriction.states,
-      };
-    }
     return {
-      restriction_type: null,
-      countries: null,
-      states: null,
+      restriction_type: courseData.location_restriction?.restriction_type || null,
+      countries: courseData.location_restriction?.countries || [],
+      states: courseData.location_restriction?.states || [],
     };
   }
 
   formatGeoLocationFields(courseData) {
-    if (courseData?.geolocation) {
-      return {
-        geoLocationName: courseData.geolocation.location_name,
-        geoLocationLng: courseData.geolocation.lng,
-        geoLocationLat: courseData.geolocation.lat,
-      };
-    }
     return {
-      geoLocationName: null,
-      geoLocationLng: null,
-      geoLocationLat: null,
+      geoLocationName: courseData.geolocation?.location_name || '',
+      geoLocationLng: courseData.geolocation?.lng || null,
+      geoLocationLat: courseData.geolocation?.lat || null,
```

Comment:
This is backend data rendering on publisher. Are you sure it is working as expected, like adding [] for fields?